### PR TITLE
Stackable items

### PR DESCRIPTION
**Pull request by CurtisGreen**
## Other
1. Fixed issues that happened with online merge of previous two commits (one added images one deleted them)
1. Fixed pesky bug where equipping an item after purchasing would duplicate it (multiple instances of the inventory object containing the items were hanging around)
1. Added .prettierignore file to ignore fonts.css file since it messes it up
1. Fixed mute button, copying an array in js will do a shallow copy, copying values individually does a deep copy

### CODE REVIEW

File: src/items/item.js [hunk 1]
```diff
@@ -57,6 +60,19 @@ export class Item extends ClickableObject {
                 }
             });
         this.displayHeight = this.sprite.displayHeight;
+
+        // Add text in top left for stackable items
+        this.numItemsText = this.scene.add
+            .text(x - 15, y - 15, this.numItems, {
+                font: "10px runescape",
+                fill: "orange",
```
Comment (willadams05) [inline src/items/item.js:68]: Mentioned on league. 1-99,999 = yellow, 100k-9999k = white, 10m+ = green

File: fonts.css [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-    url("./src/assets/fonts/runescape.svg") format("svg"), /* Legacy iOS */
-    url("./src/assets/fonts/runescape.ttf") format("truetype"), /* Safari, Android, iOS */
-    url("./src/assets/fonts/runescape.woff") format("woff"), /* Modern Browsers */
-    url("./src/assets/fonts/runescape.woff2") format("woff2"); /* Modern Browsers */
-  font-weight: normal;
-  font-style: normal;
+    font-family: "runescape";
+    src: url("./src/assets/fonts/runescape.eot"); /* IE9 Compat Modes */
+    src: url("./src/assets/fonts/runescape.eot?#iefix") format("embedded-opentype"),
+        /* IE6-IE8 */ url("./src/assets/fonts/runescape.otf") format("opentype"),
```
Comment (EthanVieira) [inline fonts.css]: Revert this change.

File: src/items/item.js [hunk 3]
```diff
@@ -112,7 +128,12 @@ export class Item extends ClickableObject {
 
     drop() {
         console.log("Drop", this.name);
-        this.destroy();
```
Comment (EthanVieira) [inline src/items/item.js]: When you drop a stackable item, it should drop the whole stack - like it does in runescape. Imagine trying to drop thousands of one item.

File: src/ui/dashboard.js [hunk 4]
```diff
@@ -118,6 +118,10 @@ export class DashboardScene extends Phaser.Scene {
                 this.inventory.obj.showInventory(true);
                 this.inventory.button.setAlpha(0.1);
             });
+        // Clear out and reinstatiate inventory
```
Comment (EthanVieira) [inline src/ui/dashboard.js]: *reinstantiate?

File: src/ui/dashboard.js [hunk 5]
```diff
@@ -318,6 +322,11 @@ export class DashboardScene extends Phaser.Scene {
 
                 this.equipment.obj.showEquipment(true);
             });
+
+        // Clear out and reinstatiate equipment
```
Comment (EthanVieira) [inline src/ui/dashboard.js]: Same spelling here

File: src/ui/inventory.js [hunk 6]
```diff
@@ -61,12 +63,32 @@ export class Inventory {
...
         for (let index = 0; index < this.playerItems.length; index++) {
-            // Check if object exists
-            if (!Object.keys(this.playerItems[index]).length) {
+            let itemExists = Object.keys(this.playerItems[index]).length;
+
+            // Check if it can stack with other items
+            if (
+                itemExists &&
+                item.stackable &&
+                this.playerItems[index].item == item.item &&
```
Comment (EthanVieira) [inline src/ui/inventory.js:72]: Can we simplify this by checking for the name?

File: src/ui/inventory.js [hunk 7]
```diff
@@ -61,12 +63,32 @@ export class Inventory {
...
+                // Update it in the cookies
+                this.playerItems[index].count += item.numItems;
+
+                return true;
+            }
+            // Check if slot is empty
+            else if (!itemExists) {
                 this.addToInventoryAtIndex(item, index, createSprite);
                 return true;
             }
```
Comment (EthanVieira) [inline src/ui/inventory.js:95]: Not sure how JS works but should we explicitly return false here?

**Review by EthanVieira — CHANGES_REQUESTED** [review]

File: fonts.css (hunk 2 above)
Comment (CurtisGreen) [reply fonts.css]: Prettier did this automatically

File: src/ui/dashboard.js (hunk 4 above)
Comment (CurtisGreen) [reply src/ui/dashboard.js]: woops

File: src/ui/inventory.js (hunk 6 above)
Comment (CurtisGreen) [reply src/ui/inventory.js:72]: No because it's checking through the cookies which has the item/type and doesn't have the "name" only so that's why I check both. You also need to check if it exists or you'll get errors on empty inventory slots

File: src/ui/inventory.js (hunk 7 above)
Comment (CurtisGreen) [reply src/ui/inventory.js:95]: I'm returning false specifically when the item cannot be added because the inventory is full. Which I do at line 76/98. Not sure what you're talking about here

File: fonts.css (hunk 2 above)
Comment (CurtisGreen) [reply fonts.css]: Do you want to revert it and then make sure prettier doesn't edit this file in the future or what?

File: src/items/item.js (hunk 3 above)
Comment (CurtisGreen) [reply src/items/item.js]: I'm considering replacing the "drop" command with "sell". What do you think about that? Also I can make it so when the item count is > 1 it can be sell 1 or sell all. Or do you think this would overinflate the user's gold (which is already pretty inflated)

File: src/items/item.js (hunk 1 above)
Comment (CurtisGreen) [reply src/items/item.js:68]: Do you want yellow or the orange that I'm using rn?

File: fonts.css (hunk 2 above)
Comment (EthanVieira) [reply fonts.css]: I want the comments to be where they should

File: src/ui/inventory.js (hunk 7 above)
Comment (EthanVieira) [reply src/ui/inventory.js:95]: I was hoodwinked by the hidden code - still not used to github prs

File: src/items/item.js (hunk 3 above)
Comment (EthanVieira) [reply src/items/item.js]: Only giving gold on monster death would help solve the inflation. Sell should probably only be when the shop is open

File: fonts.css (hunk 2 above)
Comment (CurtisGreen) [reply fonts.css]: Kk i'm going to revert this and add *.css to the newly added .prettierignore file

File: src/items/item.js (hunk 3 above)
Comment (CurtisGreen) [reply src/items/item.js]: Not sure what happened to the reponse I did. But right now the shop doesn't show inventory when it's open, if we want to sell only when the shop is open we would need to redesign the shop. If you have a good idea for a layout I can implement it. Only reason I thought of replacing "drop" with "sell" is because I don't see any reason why the user would want to drop something, unless we never want the user to sell anything

File: src/items/item.js (hunk 1 above)
Comment (CurtisGreen) [reply src/items/item.js:68]: I just tried the yellow and I think orange looks better unless you have a specific hex value of yellow you want. I implemented the white + k/green + m with 1 decimal place after each. Ex: 1.2k, 5.1m. Lmk if that's what you were thinking of

File: src/items/item.js (hunk 3 above)
Comment (EthanVieira) [reply src/items/item.js]: aight we can change to sell, but if we do that then might want a sell X item for stackable items

**Review by EthanVieira — APPROVED** [review]
